stl/inc/xstring

```diff
@@ -4295,27 +4616,46 @@ private:
         const size_type _New_capacity = _Calculate_growth(_New_size);
         auto& _Al                     = _Getal();
         const pointer _New_ptr        = _Al.allocate(_New_capacity + 1); // throws
+
+#ifdef __cpp_lib_is_constant_evaluated
+        if (_STD is_constant_evaluated()) { // Copying into uninitialized memory is technically UB, so fill it once
+            _Traits::assign(_Unfancy(_New_ptr), _New_capacity + 1, _Elem());
+        }
+#endif // __cpp_lib_is_constant_evaluated
         _My_data._Orphan_all();
         _My_data._Mysize      = _New_size;
         _My_data._Myres       = _New_capacity;
         _Elem* const _Raw_new = _Unfancy(_New_ptr);
-        if (_BUF_SIZE <= _Old_capacity) {
+#ifdef __cpp_lib_is_constant_evaluated
+        if (_STD is_constant_evaluated()) {
             const pointer _Old_ptr = _My_data._Bx._Ptr;
             _Fn(_Raw_new, _Unfancy(_Old_ptr), _Old_size, _Args...);
             _Al.deallocate(_Old_ptr, _Old_capacity + 1);
             _My_data._Bx._Ptr = _New_ptr;
-        } else {
-            _Fn(_Raw_new, _My_data._Bx._Buf, _Old_size, _Args...);
-            _Construct_in_place(_My_data._Bx._Ptr, _New_ptr);
+        } else
+#endif // __cpp_lib_is_constant_evaluated
+        {

```

Comment:
Ditto: can we combine this branch cleanly somehow with the `if (_BUF_SIZE <= _Old_capacity)` branch below, as their bodies appear to be identical?